ext/opentelemetry-ext-postgresql/src/opentelemetry/ext/postgresql/__init__.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The opentelemetry-ext-postgresql package allows tracing PostgreSQL queries made by the
+Psycopg2 library.
+"""
+
+import typing
+
+import psycopg2
+from psycopg2.extensions import cursor as pgcursor
+import wrapt
+
+from opentelemetry.context import Context
+from opentelemetry.trace import SpanKind, Tracer
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+QUERY_WRAP_METHODS = ["execute", "executemany", "callproc"]
+DATABASE_COMPONENT = "postgresql"
+DATABASE_TYPE = "sql"
+
+
+def trace_integration(tracer):
+    """Integrate with PostgreSQL Psycopg library.
+       Psycopg: http://initd.org/psycopg/
+    """
+
+    if tracer is None:
+        raise ValueError("The tracer is not provided.")
+
+    # pylint: disable=unused-argument
+    def wrap(wrapped, instance, args, kwargs):
+        """Patch Psycopg connect method.
+        """
+        connection = wrapped(*args, **kwargs)
+        connection.cursor_factory = TraceCursor
+        return connection
+
+    wrapt.wrap_function_wrapper(psycopg2, "connect", wrap)
+
+    class TraceCursor(pgcursor):
+        def __init__(self, *args, **kwargs):
+            for func in QUERY_WRAP_METHODS:
+                wrapt.wrap_function_wrapper(self, func, self.add_span)
+            super(TraceCursor, self).__init__(*args, **kwargs)
+
+        def add_span(
```

Comment:
should this be a protected method? doesn't seem like it should be callable publicly.